katdal/visdatav4.py

```diff
@@ -343,6 +295,37 @@ def __init__(self, source, ref_ant='', time_offset=0.0, centre_freq=None,
         # on selection in the process
         self.select(spw=0, subarray=0, ants=obs_ants)
 
+    @property
+    def _flags_keep(self):
+        # Reverse flag indices as np.packbits has bit 0 as the MSB (we want LSB)
+        selection = np.flipud(np.unpackbits(self._flags_select))
+        assert len(FLAG_NAMES) == len(selection), \
+            'Expected %d flag types, got %s' % (len(selection), FLAG_NAMES)
+        return [name for name, bit in zip(FLAG_NAMES, selection) if bit]
+
+    @_flags_keep.setter
+    def _flags_keep(self, names):
+        # Ensure a sequence of flag names
+        names = FLAG_NAMES if names == 'all' else \
+            names.split(',') if isinstance(names, basestring) else names
+        # Create boolean list for desired flags
+        selection = np.zeros(8, dtype=np.uint8)
+        assert len(FLAG_NAMES) == len(selection), \
+            'Expected %d flag types, got %d' % (len(selection), FLAG_NAMES)
+        for name in names:
+            try:
+                selection[FLAG_NAMES.index(name)] = 1
+            except ValueError:
+                logger.warning("%r is not a legitimate flag type, "
+                               "supported ones are %s", name, FLAG_NAMES)
+        # Pack index list into bit mask
+        # Reverse flag indices as np.packbits has bit 0 as the MSB (we want LSB)
+        flagmask = np.packbits(np.flipud(selection))
+        if not flagmask:
+            logger.warning('No valid flags were selected - setting all flags '
+                           'to False by default')
+        self._flags_select = flagmask
+
```

Comment:
Is this common code that could be pulled out into its own class and shared with v3? Maybe it will make more sense to do so once v3 has been dask-ified though.

Reply:
Yes, agreed.